docs/architecture.md

```diff
@@ -0,0 +1,218 @@
+# Skenario Architecture
+
+This document introduces the architecture of Skenario, a simulator for
+HPA and VPA.
+
+Skenario is an open-source simulator and interface for the HPA and VPA autoscalers.
+Skenario provides a simulated cluster with a web UI.  The user configures an HPA or VPA object, 
+selects a predefined traffic profile and receives a collection of metrics which are plotted on 
+the screen showing what happened during the simulation.
+
+##Terms
+
+* `Implementation` - the autoscaling system under test.
+* `Scenario` - a set of parameters which include input traffic, cluster configuration and limits, 
+  and an Implementation under test.
+* `Simulation Environment (or just Simulation)` - the core machinery, instantiated once per Scenario.
+* `Autoscaler Interface` - the interface over which the controller (autoscaler) passes input parameters 
+  and over which the simulation passes back output metrics.
+* `Plugins` - autoscalers wrapped in order to implement the autoscaler interface and to be driven 
+  deterministically by an injected clock.
+
+## Skenario architecture diagram
+
+```   
+            
+                            plugin-k8s --- Kubernetes
+                sk-plugin /                  /horizontal.go
+         sk-plugin       / 
+simulator <--> dispatcher 
+                         \ 
+                sk-plugin \
+                          plugin-k8s-vpa --- Autoscaler
+                                               /recommender.go 
+
+```
+
+* `simulator` - the core Simulation Environment
+* `dispatcher` - responsible for plugin management
+* `plugin-k8s, plugin-k8s-vpa` - autoscalers (hpa and vpa)wrapped in order to implement 
+  the autoscaler interface and to be driven deterministically by an injected clock
+* `sk-plugin` - Autoscaler Interface 
```

Comment:
I think this is just `plugin` now, isn't it?

Reply:
Sure, you're right. Renamed it.